src/components/MultiHopTrade/components/LimitOrder/hooks/useAllowanceApproval.tsx

```diff
@@ -0,0 +1,102 @@
+import { assertGetViemClient } from '@shapeshiftoss/contracts'
+import { COW_SWAP_VAULT_RELAYER_ADDRESS } from '@shapeshiftoss/swapper'
+import { useMutation } from '@tanstack/react-query'
+import { useEffect, useMemo } from 'react'
+import { reactQueries } from 'react-queries'
+import type { Hash } from 'viem'
+import {
+  AllowanceType,
+  getApprovalAmountCryptoBaseUnit,
+  useApprovalFees,
+} from 'hooks/queries/useApprovalFees'
+import { useIsAllowanceApprovalRequired } from 'hooks/queries/useIsAllowanceApprovalRequired'
+import { useWallet } from 'hooks/useWallet/useWallet'
+import type { LimitOrderActiveQuote } from 'state/slices/limitOrderSlice/types'
+import { selectAccountNumberByAccountId } from 'state/slices/selectors'
+import { useAppSelector } from 'state/store'
+
+type UseAllowanceApprovalProps = {
+  activeQuote: LimitOrderActiveQuote
+  setTxHash: (txHash: string) => void
+  feeQueryEnabled: boolean
+  isInitiallyRequired: boolean
+  onMutate: () => void
+  onError: (err: Error) => void
+  onSuccess: () => void
+}
+
+// handles allowance approval tx execution, fees, and state orchestration
+export const useAllowanceApproval = ({
+  activeQuote,
+  setTxHash,
+  feeQueryEnabled,
+  isInitiallyRequired,
+  onMutate,
+  onError,
+  onSuccess,
+}: UseAllowanceApprovalProps) => {
+  const wallet = useWallet().state.wallet ?? undefined
+
+  const accountNumberFilter = useMemo(
+    () => ({ accountId: activeQuote.params.accountId }),
+    [activeQuote.params.accountId],
+  )
+  const accountNumber = useAppSelector(state =>
+    selectAccountNumberByAccountId(state, accountNumberFilter),
+  )
+
+  const { allowanceCryptoBaseUnitResult, isAllowanceApprovalRequired } =
+    useIsAllowanceApprovalRequired({
+      amountCryptoBaseUnit: activeQuote.params.sellAmountCryptoBaseUnit,
+      assetId: activeQuote.params.sellAssetId,
+      from: activeQuote.params.sellAccountAddress,
+      spender: COW_SWAP_VAULT_RELAYER_ADDRESS,
+    })
+
+  const { evmFeesResult } = useApprovalFees({
+    amountCryptoBaseUnit: activeQuote.params.sellAmountCryptoBaseUnit,
```

Comment:
preferably-blocking: Note, this will produce a suboptimal state in case the user is within USDT reset unsupported flow: this will try and simulate a Tx for the actual allowance we want, which will fail.
As a result, this will be undefined, which means we will hit this condition and show an errored state which "Insufficient funds for gas", which is incorrect:

https://github.com/shapeshift/web/blob/9fff484a3351dbca5e8d4abfc46dcbe801c15833/src/components/MultiHopTrade/components/LimitOrder/components/AllowanceApproval.tsx#L136-L138